src/asciidoctor-core-api.js

```diff
@@ -365,8 +370,28 @@ AbstractNode.$$proto.getAttributes = function () {
  * @memberof AbstractNode
  */
 AbstractNode.$$proto.getAttribute = function (name, defaultValue, inherit) {
-  var value = this.$attr(name, defaultValue, inherit);
-  return value === Opal.nil ? undefined : value;
+  if (inherit == null)
```

Comment:
I have the feeling that we should delegate more on Asciidoctor core (ie. use the method `attr` to get the attribute).

Reply:
As a matter of fact, I had to factor out that logic from $attr since, as far as I understand the code of that later, it assumes attribute names are strings. But maybe that was `$attr` I should have modified to properly handle non-string keys?